### ISSUE TITLE
`jj <alias> --help` works, but `jj help <alias>` does not

### ISSUE THREAD

**Issue by torquestomp** [martinvonz/jj#2866]
Example: Add an alias `fetch = git fetch` in your config, run `jj help fetch`.  You get `error: unrecognized subcommand 'fetch'`

**Comment by ilyagr** [comment]
I wouldn't say that `jj <alias> --help` works properly, either.

E.g., if I have an aliase `d = ["diff", "-s"]`, `jj d --help` is the same as `jj diff --help`. I'd prefer if `jj d --help` told me that `d` is an alias. We could *also* make it print `jj diff --help` afterwards, or we could not do that.

**Comment by torquestomp** [comment]
I was hoping this would be a quick in-and-out fix, but it appears that the way `help` is actually invoked is deep in the internals of clap, so it's not as simple as I hoped.  I may come back to this later but I'll unassign for now.

**Comment by martinvonz** [comment]
> but it appears that the way `help` is actually invoked is deep in the internals of clap

Yes, I suspect we'll need to take over handling of all help messages in order to fix this.

**Comment by jyn514** [comment]
Rather than implementing `help` manually, it might make sense to go through `visible_alias`, which makes help work out of the box without having to reimplement it: https://docs.rs/clap/latest/clap/struct.Command.html#method.visible_alias
(i took a look at reimplementing help and it looks decidedly non-trivial, there's lots of digging through clap's reflection API.)

**Comment by jyn514** [comment]
ah, i see, aliases can be multiple commands, they don't correspond 1-1 with a subcommand. so visible_alias won't work.

**Comment by jyn514** [comment]
for anyone picking this up in the future - the way clap does this internally is https://docs.rs/clap_builder/4.4.18/src/clap_builder/builder/command.rs.html#4474-4487

**Comment by joyously** [comment]
Is there a place where the "not found" error can be checked, and a list of aliases could be listed instead?

**Comment by jyn514** [comment]
@joyously do you mean the "unrecognized subcommand" error? yes, but that place is in clap itself: https://github.com/jyn514/clap/blob/a7e04a53e4a9e310663a0cdb5bfb0753d84b82bd/clap_builder/src/error/format.rs#L529

**Comment by LucasWagler** [comment]
Related: `jj b --help` displays the top-level help text, instead of the `bookmark` help text. Other aliases seem to behave similarly.

This seems to contradict behavior mentioned in this previous comment, as the help page now differs between the full subcommand and any aliases to that subcommand:
>E.g., if I have an alias aliases.d = ["diff", "-s"], jj d --help is currently the same as jj diff --help.

**Comment by ilyagr** [comment]
Copying from Discord, I can't reproduce the `jj b --help` problem on my Mac. It seems that Lucas and @icorbrey both can, so things are weird.

<details>

```console
$ JJ_CONFIG= jj b -h   # Same with or without JJ_CONFIG=
Manage bookmarks [default alias: b]

Usage: jj-darwin-aarch64 bookmark [OPTIONS] <COMMAND>

Commands:
  create   Create a new bookmark [aliases: c]
  delete   Delete an existing bookmark and propagate the deletion to remotes on the next push [aliases: d]
  forget   Forget a bookmark without marking it as a deletion to be pushed [aliases: f]
  list     List bookmarks and their targets [aliases: l]
  move     Move existing bookmarks to target revision [aliases: m]
  rename   Rename `old` bookmark name to `new` bookmark name [aliases: r]
  set      Create or update a bookmark to point to a certain commit [aliases: s]
  track    Start tracking given remote bookmarks [aliases: t]
  untrack  Stop tracking given remote bookmarks

Options:
  -h, --help  Print help (see more with '--help')

Global Options:
  -R, --repository <REPOSITORY>      Path to repository to operate on
<...REDACTED...>

$ JJ_CONFIG= jj b --help | head
Manage bookmarks [default alias: b]

See [`jj help -k bookmarks`] for more information.

[`jj help -k bookmarks`]: https://jj-vcs.github.io/jj/latest/bookmarks

Usage: jj-darwin-aarch64 bookmark [OPTIONS] <COMMAND>

Commands:
  create   Create a new bookmark [aliases: c]

$ JJ_CONFIG= jj help b
error: unrecognized subcommand 'b'

  tip: some similar subcommands exist: 'bookmark', 'abandon', 'backout', 'absorb', 'obslog', 'rebase'

Usage: jj-darwin-aarch64 [OPTIONS] <COMMAND>

For more information, try '--help'.
```

</details>

**Comment by LucasWagler** [comment]
@ilyagr A quick Nushell vs. zsh comparison reveals this issue is likely Nushell-specific. I'll dig in & maybe create a separate issue.

Edit: `jj b "--help"` in Nushell behaves properly (explicitly using a string arg for "--help"), so this definitely seems Nushell-related

Edit 2: New issue created #6945
